### ISSUE TITLE
[explicit-function-return-type] ignore implicit void

### ISSUE THREAD

**Issue by sam-s4s** [typescript-eslint/typescript-eslint#3534]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-function-return-type": ["error"]
  }
}
```

```TS
myFunction() {
    this.doSomething();
}
```

**Expected Result**
(when option is enabled) No warning

**Actual Result**
Missing return type on function  @typescript-eslint/explicit-function-return-type

**Additional Info**

I am surprised to find that there is no option for this rule to ignore implicit void returns. To my knowledge, it's pretty standard practice not to specify the return type of a function when there is no return at all.

eg. in our codebase we have about 3,000 hits of the "@typescript-eslint/explicit-function-return-type" rule, of which almost all of them are for void returns.

I'd love to have this rule, as I feel it is very valuable to specify returns when there are returns. But at present, I cannot enable this rule.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.23.0` |
| `@typescript-eslint/parser`        | `4.23.0` |
| `TypeScript`                       | `4.2.4` |
| `ESLint`                           | `7.27.0` |
| `node`                             | `14.17.0` |

**Comment by bradzacher** [comment]
Nope!
Use the search next time: #50
